framework/src/main/java/org/checkerframework/framework/flow/CFAbstractAnalysis.java

```diff
@@ -88,6 +88,13 @@ public CFAbstractAnalysis(
                 factory.getQualifierHierarchy().numberOfIterationsBeforeWidening());
     }
 
+    /** Reset the analysis so it can be reused. */
```

Comment:
Why provide a reset mechanism only for `CFAbstractAnalysis` instead of for all kinds of `Analysis`?
Wouldn't it be better to put the body of `org.checkerframework.dataflow.analysis.Analysis.performAnalysis(ControlFlowGraph)` into a helper method and have a try-catch block around the invocation of that helper method and set/reset `isRunning` there?

Reply:
>Wouldn't it be better to put the body of org.checkerframework.dataflow.analysis.Analysis.performAnalysis(ControlFlowGraph) into a helper method and have a try-catch block around the invocation of that helper method and set/reset isRunning there?

I can do that, but this problem happens if any of the code crashes GenericAnnotatedTypeFactory#performFlowAnalysis, we still need to sit isRunning to false in that method.

Reply:
The other alternative is to remove the `assert !isRunning` from Analysis#performAnalysis.